internal/provision/provisionwatchers.go

```diff
@@ -96,3 +118,42 @@ func LoadProvisionWatchers(path string, dic *di.Container) errors.EdgeX {
 	_, edgexErr := pwc.Add(ctx, addProvisionWatchersReq)
 	return edgexErr
 }
+
+func processFile(lc logger.LoggingClient, path string, secretProvider interfaces.SecretProvider, addProvisionWatchersReq []requests.AddProvisionWatcherRequest) {
+	var watcher dtos.ProvisionWatcher
+	data, err := file.Load(path, timeout, secretProvider)
+	if err != nil {
+		lc.Errorf("Failed to read %s: %v", path, err)
+		return
+	}
+
+	if strings.HasSuffix(path, yamlExt) || strings.HasSuffix(path, ymlExt) {
+		err = yaml.Unmarshal(data, &watcher)
+		if err != nil {
+			lc.Errorf("Failed to YAML decode %s: %v", path, err)
+			return
+		}
+	} else if strings.HasSuffix(path, jsonExt) {
+		err := json.Unmarshal(data, &watcher)
+		if err != nil {
+			lc.Errorf("Failed to JSON decode %s: %v", path, err)
```

Comment:
Don't like this for local, but URL would leak creds
```suggestion
			lc.Errorf("Failed to JSON decode Provision Watcher: %v", err)
```